core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Parameter.kt

```diff
@@ -49,23 +29,32 @@ class Parameter<T : ValidType> : TypeExpression<T> {
                 parameters = mapOf(unnamedParameterCount to value),
             )
         }
-
         else -> DopeQuery(
             queryString = "\$$parameterName",
             parameters = mapOf(parameterName to value),
         )
     }
 }
 
-fun Number.asParameter(parameterName: String? = null) = Parameter<NumberType>(this, parameterName)
-fun String.asParameter(parameterName: String? = null) = Parameter<StringType>(this, parameterName)
-fun Boolean.asParameter(parameterName: String? = null) = Parameter<BooleanType>(this, parameterName)
+class NumberParameter(value: Number, parameterName: String?) : Parameter<NumberType>(value, parameterName)
+
+class StringParameter(value: String, parameterName: String?) : Parameter<StringType>(value, parameterName)
+
+class BooleanParameter(value: Boolean, parameterName: String?) : Parameter<BooleanType>(value, parameterName)
+
+class ArrayParameter<T : ValidType>(value: Collection<Any>, parameterName: String?) : Parameter<ArrayType<T>>(value, parameterName)
+
+fun Number.asParameter(parameterName: String? = null) = NumberParameter(this, parameterName)
+
+fun String.asParameter(parameterName: String? = null) = StringParameter(this, parameterName)
+
+fun Boolean.asParameter(parameterName: String? = null) = BooleanParameter(this, parameterName)
 
 @JvmName("numberCollectionAsParameter")
-fun Collection<Number>.asParameter(parameterName: String? = null) = Parameter<ArrayType<NumberType>>(this, parameterName)
+fun Collection<Number>.asParameter(parameterName: String? = null) = ArrayParameter<NumberType>(this, parameterName)
```

Comment:
What if it is a list of a list?

Reply:
I think that's not part of this ticket. please create a new ticket for nested lists.